txircd/modules/extra/webirc.py

```diff
@@ -0,0 +1,63 @@
+from twisted.plugin import IPlugin
+from twisted.words.protocols import irc
+from txircd.config import ConfigValidationError
+from txircd.module_interface import Command, ICommand, IModuleData, ModuleData
+from txircd.utils import isValidHost, resolveHost
+from zope.interface import implements
+
+irc.ERR_ALREADYREGISTERED = "462"
+
+class WebIRC(ModuleData, Command):
+	implements(IPlugin, IModuleData, ICommand)
+
+	name = "WebIRC"
+	forRegistered = False
+
+	def actions(self):
+		return [ ("commandpermission-WEBIRC", 10, self.checkSourceAndPass) ]
+
+	def userCommands(self):
+		return [ ("WEBIRC", 1, self) ]
+
+	def verifyConfig(self, config):
+		if "webirc_allowed_sources" in config:
+			if not isinstance(config["webirc_allowed_sources"], dict):
+				raise ConfigValidationError("webirc_allowed_sources", "value must be a dictionary")
+			for ip, password in config["webirc_allowed_sources"].iteritems():
+				if not isinstance(ip, basestring):
+					raise ConfigValidationError("webirc_allowed_sources", "ip value must be a string")
+				if not isinstance(password, basestring):
+					raise ConfigValidationError("webirc_allowed_sources", "password value must be a string")
+
+	def checkSourceAndPass(self, user, data):
```

Comment:
You missed a return at the end of the function.

(Explicitly `return None` to make it obvious you intended that and didn't forget a different return value.)